services/src/main/java/io/scalecube/services/Microservices.java

```diff
@@ -322,7 +326,12 @@ private GatewayBootstrap addConfig(GatewayConfig config) {
     }
 
     private Mono<Void> shutdown() {
-      return Flux.fromIterable(gatewayInstances.values()).flatMap(Gateway::stop).then();
+      return Mono.defer(
+          () ->
+              gatewayInstances != null && !gatewayInstances.isEmpty()
```

Comment:
In accordance with similar changes maybe better to use `Optional`?

Reply:
@dmytro-lazebnyi nope